avro-annotations/src/main/java/io/micronaut/avro/AvroSchemaSource.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2017-2025 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.avro;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * An annotation that loads the Avro schema from the resources' folder.
+ * This annotation can be applied to classes.
+ *
+ * @since 1.0
+ * @author Ali Linaboui
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+public @interface AvroSchemaSource {
+
+    /**
+     * Specifies the name of the Avro schema file to be loaded.
+     * like ("salamander-schema.avsc")
+     *
+     * @return the name of the Avro schema file.
+     */
+    String value();
```

Comment:
should support `classpath:`, `file:` and possibly `url:` prefixes. The former 2 should use the resoureloader infra